modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,509 @@
+      copyMeshProperty<bool>("is_control_drum_meta", name() + "_fpg");
+      copyMeshProperty<Real>("pattern_pitch_meta", name() + "_fpg");
+    }
+    std::string build_mesh_name = name() + "_delbds";
+    {
+      // Invoke BoundaryDeletionGenerator to delete extra sidesets created by
+      // AdvancedConcentricCircleGenerator
+      auto params = _app.getFactory().getValidParams("BoundaryDeletionGenerator");
+
+      params.set<MeshGeneratorName>("input") = name() + "_fpg";
+      params.set<std::vector<BoundaryName>>("boundary_names") = {"0", "2"};
+
+      addMeshSubgenerator("BoundaryDeletionGenerator", build_mesh_name, params);
+    }
+    if (_extrude && _mesh_dimensions == 3)
+    {
+      std::vector<Real> axial_boundaries =
+          getReactorParam<std::vector<Real>>(RGMB::axial_mesh_sizes);
+      const auto top_boundary = getReactorParam<boundary_id_type>(RGMB::top_boundary_id);
+      const auto bottom_boundary = getReactorParam<boundary_id_type>(RGMB::bottom_boundary_id);
+      {
+        auto params = _app.getFactory().getValidParams("AdvancedExtruderGenerator");
+
+        params.set<MeshGeneratorName>("input") = build_mesh_name;
+        params.set<Point>("direction") = Point(0, 0, 1);
+        params.set<std::vector<unsigned int>>("num_layers") =
+            getReactorParam<std::vector<unsigned int>>(RGMB::axial_mesh_intervals);
+        params.set<std::vector<Real>>("heights") = axial_boundaries;
+        params.set<boundary_id_type>("bottom_boundary") = bottom_boundary;
+        params.set<boundary_id_type>("top_boundary") = top_boundary;
+
+        addMeshSubgenerator("AdvancedExtruderGenerator", name() + "_extruded", params);
+      }
+
+      {
+        auto params = _app.getFactory().getValidParams("RenameBoundaryGenerator");
+
+        params.set<MeshGeneratorName>("input") = name() + "_extruded";
+        params.set<std::vector<BoundaryName>>("old_boundary") = {
+            std::to_string(top_boundary),
+            std::to_string(bottom_boundary)}; // hard coded boundary IDs in patterned mesh generator
+        params.set<std::vector<BoundaryName>>("new_boundary") = {"top", "bottom"};
+
+        addMeshSubgenerator("RenameBoundaryGenerator", name() + "_change_plane_name", params);
+      }
+
+      {
+        auto params = _app.getFactory().getValidParams("PlaneIDMeshGenerator");
+
+        params.set<MeshGeneratorName>("input") = name() + "_change_plane_name";
+
+        std::vector<Real> plane_heights{0};
+        for (Real z : axial_boundaries)
+          plane_heights.push_back(z + plane_heights.back());
+
+        params.set<std::vector<Real>>("plane_coordinates") = plane_heights;
+
+        std::string plane_id_name = "plane_id";
+        params.set<std::string>("id_name") = "plane_id";
+
+        build_mesh_name = name() + "_extrudedIDs";
+        addMeshSubgenerator("PlaneIDMeshGenerator", build_mesh_name, params);
+      }
+    }
```

Comment:
these 3 generators for extrusion + rename + planes, dont we do this similarly elsewhere? Maybe we should have a util

Reply:
alternatively, how often do we extrude only a single control drum? shouldnt we always just extrude the core?
we could remove the option altogether. just tell the user to use a CoreMeshGenerator if they really need it

Reply:
I decided to add the extrusion routines here for CD Mesh Generator just to be consistent with all other RGMB mesh generators. To simplify this code, I moved all extrusion + rename + plane ID mesh subgenerator calls to the base class